cmd/util/testutils/testutils.go

```diff
@@ -18,25 +18,78 @@
 package testutils
 
 import (
-	"net/http"
-	"os"
+	"bytes"
+	"fmt"
+	"io"
+	"testing"
 
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/api/mock"
 	"github.com/elastic/cloud-sdk-go/pkg/output"
 
 	"github.com/elastic/ecctl/pkg/ecctl"
-	"github.com/elastic/ecctl/pkg/util"
 )
 
-// MockInitApp initiates a mocked app
-func MockInitApp() error {
-	var config = ecctl.Config{
-		Client:       new(http.Client),
-		OutputDevice: output.NewDevice(os.Stdout),
-		ErrorDevice:  os.Stderr,
-		Output:       "json",
-		Host:         "http://somehost",
-		APIKey:       "helloiamakey",
+const (
+	defaultOutputFormat = "json"
+	defaultAPIKey       = "dummy"
+	defaultRegion       = "ece-region"
+)
+
+// MockCfg represents a small and targeted amount of `ecctl.Config` options
+// aimed at making mocking convenient and easy.
+type MockCfg struct {
+	Responses    []mock.Response
+	Out          io.Writer
+	Err          io.Writer
+	OutputFormat string
+	Format       string
+	Region       string
+
+	Force   bool
+	Verbose bool
+}
+
+func fillDefaults(cfg MockCfg) MockCfg {
+	if cfg.OutputFormat == "" {
+		cfg.OutputFormat = defaultOutputFormat
+	}
+
+	if cfg.Region == "" {
+		cfg.Region = defaultRegion
+	}
+
+	if cfg.Err == nil {
+		cfg.Err = new(bytes.Buffer)
+	}
+
+	if cfg.Out == nil {
+		cfg.Out = new(bytes.Buffer)
+	}
+
+	return cfg
+}
+
+func newConfig(cfg MockCfg) ecctl.Config {
+	cfg = fillDefaults(cfg)
+	return ecctl.Config{
+		Client:       mock.NewClient(cfg.Responses...),
+		OutputDevice: output.NewDevice(cfg.Out),
+		ErrorDevice:  cfg.Err,
+		Output:       cfg.OutputFormat,
+		Format:       cfg.Format,
+		Host:         fmt.Sprintf("https://%s", api.DefaultMockHost),
+		APIKey:       defaultAPIKey,
+		Force:        cfg.Force,
+		Verbose:      cfg.Verbose,
+	}
+}
+
+// MockApp initiates a mocked app from a MockCfg.
+func MockApp(t *testing.T, cfg MockCfg) func() {
```

Comment:
This doesn't need to be exported any more, does it?

Reply:
Not really no, but it'd allow us to call it externally if we need? although I'm not sure we need to anymore

Reply:
I don't think we need to any more 🤔 If we really need it we can export it later down the track. I also wouldn't want someone to find this function and skip the `TestCommand` function or whatever. What do you think?

Reply:
yeah Ok, I'll hide this function